bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -611,7 +544,7 @@ int getReturnRc(int rc) {
         }
     }
 
-    void scheduleBookieHealthCheckIfEnabled() {
+    void scheduleBookieHealthCheckIfEnabled(ClientConfiguration conf) {
```

Comment:
I'd prefer to change this in a second pass where I make BookKeeper use ClientContext too. This way you can use a fully fledged bk client with all the outcalls mocked. It would also get rid of the need for a separate MockBookKeeper implementation.